design/readonly-variables.md

```diff
@@ -37,9 +37,6 @@ All of the following should be readonly:
 - Queue-time variables
 - Task- and script-created variables with a new setting, `isReadonly=true`
 
-Additionally, agents should no longer overwrite the following important environment variables, even if there's a pipeline variable with the corresponding name:
-- `TODO`
```

Comment:
Why did you end up cutting this? Not super opinionated either way, but I think I'd just like to understand the rationale - I think I see where this could be helpful, but its also definitely a challenging/weird line to draw.

Reply:
FWIW - everything else looks good, was just gating my approval on this

Reply:
Great question. I started trying to enumerate the variables we should "protect" on each platform. I eventually realized two things:
- It's an arms race, albeit a slow one. Apple just moved from `bash` to `zsh`, and there are innumerable shell options on Linux. What environment variables are meaningful to each one, and how do we know when we're in a state to honor that?
- There are narrow legitimate use cases for overriding an environment variable. `GOPATH`/`GOROOT` was my go-to example (though it's less relevant now). I don't feel qualified at this moment to enumerate those use cases and carve out exceptions for them.

Basically, this was always kind of a sideshow to the notion of readonly pipeline variables. Instead of trying to poorly design it in a footnote here, if we care enough, we should open a real feature & design for it.

Reply:
Cool, that makes sense to me!